runtime/src/accounts_index_storage.rs

```diff
@@ -0,0 +1,65 @@
+use crate::bucket_map_holder::BucketMapHolder;
+use crate::waitable_condvar::WaitableCondvar;
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    thread::{Builder, JoinHandle},
+};
+
+// eventually hold the bucket map
+// Also manages the lifetime of the background processing threads.
+//  When this instance is dropped, it will drop the bucket map and cleanup
+//  and it will stop all the background threads and join them.
+
+#[derive(Debug, Default)]
+pub struct AccountsIndexStorage {
+    // for managing the bg threads
+    exit: Arc<AtomicBool>,
+    wait: Arc<WaitableCondvar>,
+    handle: Option<JoinHandle<()>>,
+
+    // eventually the backing storage
+    storage: Arc<BucketMapHolder>,
+}
+
+impl Drop for AccountsIndexStorage {
+    fn drop(&mut self) {
+        self.exit.store(true, Ordering::Relaxed);
+        self.wait.notify_all();
+        if let Some(x) = self.handle.take() {
+            x.join().unwrap()
+        }
+    }
+}
```

Comment:
Does the order that these stores are visible to other threads matter? If so, then the `exit` ordering probably needs to change. If the other threads can handle any order of these stores being visible, then this is all good.

Reply:
brooks and I discussed in slack. worst case is tests take 10s. this code will be reworked once we have more bg threads that are actually doing work.